.github/workflows/gh-pages.yml

```diff
@@ -1,35 +1,25 @@
-# This is a basic workflow to help you get started with Actions
-
-name: github-pages
-
-# Controls when the action will run. Triggers the workflow on push or pull request
-# events but only for the master branch
+name: Deploy to GitHub Pages
 on:
   push:
     branches:
-      - development
+      - main
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
   deploy:
-    # The type of runner that the job will run on
-    # Not latest because peaceiris/actions-gh-pages@v3 requires this version
-    runs-on: ubuntu-18.04 
+    runs-on: ubuntu-latest
+
     steps:
       - name: Checkout
         uses: actions/checkout@v2
 
-      - name: Setup mdBook
-        uses: peaceiris/actions-mdbook@v1
-        with:
-          mdbook-version: 'latest'
-
-      - name: Build
-        run: mdbook build
+      - name: Install and Build
+        run: |
+          yarn
+          yarn build
 
       - name: Deploy
-        uses: peaceiris/actions-gh-pages@v3
+        uses: JamesIves/github-pages-deploy-action@3.5.9
         with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          publish_dir: ./book
-          publish_branch: master
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
@igaray as a requirement to build and deploy this project when merged to "main" branch we will need to provide a write access token as secret to this repository in order for this action to push to `gh-pages` branch.